crates/ide/src/navigation_target.rs

```diff
@@ -59,12 +59,46 @@ impl fmt::Debug for NavigationTarget {
         }
         f.field("file_id", &self.file_id).field("full_range", &self.full_range);
         opt!(focus_range);
-        f.field("name", &self.name);
+        f.field("name", self.name.inner());
         opt!(kind container_name description docs);
         f.finish()
     }
 }
 
+#[derive(Clone, PartialEq, Eq, Hash, PartialOrd)]
+pub enum NavigationTargetName {
+    /// Name of a `NavigationTarget`
+    Original(SmolStr),
+    /// use : Alias(alias , original)
+    /// At times we use aliases in those cases we must store
+    /// what the original name of a `NavigationTarget` is.
+    Alias(SmolStr, SmolStr),
+}
+
+impl Ord for NavigationTargetName {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.inner().cmp(other.inner())
+    }
+}
+
+impl NavigationTargetName {
+    pub fn inner(&self) -> &SmolStr {
```

Comment:
Naming, should be `as_inner` https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv